asar-teensy-firmware/TestScripts/Combined/MotorandProximity/Proximity_Sensor.cpp

```diff
@@ -0,0 +1,48 @@
+/* Proximity_Sensor.cpp
+ *  Author: Aaron Mallabar
+ *  Date:   February 19, 2018
+ 	Last Update: February 19, 2018
+ *  Description: Low Level Motor Control
+ *  
+ */
+ 
+
+#include "Arduino.h"
+#include "Proximity_Sensor.h"
+
+
+namespace ASAR
+{
+Proximity_Sensor::Proximity_Sensor()
+{
+	pinMode(this->trigPin, OUTPUT);
+	pinMode(this->echoPin, INPUT);
+}
+
+Proximity_Sensor::~Proximity_Sensor()
+{
+	//Empty Destructor
+}
+
+
+
+
+int Proximity_Sensor::CheckProximity()
+{
+  int duration = 0;  //Time taken to receive signal back (proportional to distance)
+  double distance = 0; //Distance away from an object
+
+ /*
+  * Sends out a pulse
+  */
+  digitalWrite(trigPin, LOW);  
+  delayMicroseconds(2);
+  digitalWrite(trigPin, HIGH);
+  delayMicroseconds(10);
+  digitalWrite(trigPin, LOW);
+
+  duration = pulseIn(echoPin, HIGH); //waits to receive pulse, and times it
+  distance = duration*0.034/2; //convert to a distance, (possibly cm?)
```

Comment:
Magic numbers.

Reply:
added const int SPEED_OF_SOUND //in cm
2 is just to take into account that it travels there and back, I dont think its necessary to make it a variable